CorsixTH/Src/th_map.cpp

```diff
@@ -1017,12 +1017,50 @@ void level_map::draw(render_target* pCanvas, int iScreenX, int iScreenY,
       continue;
     }
 
-    for (map_scanline_iterator itrNode(
-             itrNode1, map_scanline_iterator_direction::backward, iCanvasX,
-             iCanvasY);
-         itrNode; ++itrNode) {
+    map_scanline_iterator itrNext(
+        itrNode1, map_scanline_iterator_direction::forward, iCanvasX, iCanvasY);
+
+    if (!bFirst) {
+      // since the scanline count from one THMapScanlineIterator to
+      // another can differ synchronization between the current iterator
+      // and the former one is needed
+      if (itrNext.x() < -64) {
+        ++itrNext;
+      }
+      while (formerIterator.x() < itrNext.x()) {
+        ++formerIterator;
+      }
+    }
+    bool bPreviousTileNeedsRedraw = false;
+    bool predrawNextWestWall = false;
+    for (; itrNext;) {
+      map_scanline_iterator itrNode = itrNext;
+      ++itrNext;
+      bool bNeedsRedraw = false;
       int iH;
-      unsigned int iBlock = itrNode->iBlock[1];
+      // If we drew the next west wall in the previous iteration, skip it in
+      // this iteration.
+      bool skipWestWall = predrawNextWestWall;
+      // If we have a next tile, we are not at the maximum x value so it is safe
+      // to get the adjacent tile.
+      const map_tile* adjacent =
+          itrNext ? itrNode.get_adjacent_tile() : nullptr;
```

Comment:
Just itrNext.get_tile(), no reason to grab an adjacent one from itrNode.

Reply:
See response above, the scanline iterator steps to x + 1, y - 1, to maintain the same vertical position on screen. The tile I am interested in is the one at x + 1 to see if the wall on the current tile continues in that direction.